src/main/scala/org/broadinstitute/dsde/firecloud/service/FireCloudDirectives.scala

```diff
@@ -22,37 +22,47 @@ trait FireCloudDirectives extends spray.routing.Directives with PerRequestCreato
   def respondWithJSON = respondWithMediaType(`application/json`)
 
   def passthrough(unencodedPath: String, methods: HttpMethod*) = methods map { inMethod =>
-    val outMethod = new RequestBuilder(inMethod)
+    generateExternalHttpPerRequestForMethod(requestCompression = false, unencodedPath, inMethod)
+  } reduce (_ ~ _)
 
-    val path = encodeUri(unencodedPath)
+  def passthrough(requestCompression: Boolean, unencodedPath: String, methods: HttpMethod*) = methods map { inMethod =>
+    generateExternalHttpPerRequestForMethod(requestCompression, unencodedPath, inMethod)
+  } reduce (_ ~ _)
+
+  def passthroughAllPaths(ourEndpointPath: String, targetEndpointUrl: String, requestCompression: Option[Boolean] = Some(false)) = pathPrefix(ourEndpointPath) {
```

Comment:
could you make the requestCompression argument be `requestCompression: Boolean = false`, instead of an `Option[Boolean]`? Then you could remove the match/case, and just pass the value through. Since the `passthrough()` signature that omits the requestCompression argument defaults to false anyway, this should be equivalent functionality and less code.